File: schemas/Message.go
package schemas

type Message struct {
	Role      string     `json:"role"`                 // user || assistant || tool || system
	Content   string     `json:"content"`
	ToolCalls []ToolCall `json:"tool_calls,omitempty"` // empty if content present
	ToolName  string     `json:"tool_name,omitempty"`
}

File: source/ui/web/public/types/Message.mjs

export const Message = function() {

	this.Role    = "";
	this.Content = "";
	this.Created = new Date();

};

Message.from = (data) => {

	let message = new Message();

	message.Role    = data["role"]    || "";
	message.Content = data["content"] || "";
	message.Created = new Date((data["created"] || "").replace(" ", "T"));

	return message;

};
